src/test/java/com/uber/cadence/workflow/WorkflowTest.java

```diff
@@ -2814,6 +2817,51 @@ public void testSideEffect() {
     tracer.setExpected("sideEffect", "sleep PT1S", "executeActivity customActivity1");
   }
 
+  private static final Map<String, Long> mutableSideEffectValue =
```

Comment:
Seems using a queue will make this much simpler? Then we can get rid of all the sleep stuffs.

Reply:
So i tried this idea. The interesting thing is that it only works if I retain the sleep in the workflow code. If that is removed then handleMutableSideEffectMarker will not be called and mutableSideEffect will always think the value does not exist.

Reply:
Interesting. It looks like it indicates that the current implementation is broken when the mutableSideEffect is called more than once per decision. Let me investigate.